pkg/util/util_test.go

```diff
@@ -3,9 +3,228 @@ package util
+			}
+		})
+	}
+}
+
+func TestGetComponentDir(t *testing.T) {
+	type args struct {
+		path      string
+		paramType ComponentCreateType
+	}
+	tests := []struct {
+		testName string
+		args     args
+		want     string
+		wantErr  bool
+	}{
+		{
+			testName: "Case: Git URL",
+			args: args{
+				paramType: GIT,
+				path:      "https://github.com/openshift/nodejs-ex.git",
+			},
+			want:    "nodejs-ex",
+			wantErr: false,
+		},
+		{
+			testName: "Case: Source Path",
+			args: args{
+				paramType: SOURCE,
+				path:      "./testing",
+			},
+			wantErr: false,
+			want:    "testing",
+		},
+		{
+			testName: "Case: Binary path",
+			args: args{
+				paramType: BINARY,
+				path:      "./testing.war",
+			},
+			wantErr: false,
+			want:    "testing",
+		},
+		{
+			testName: "Case: No clue of any component",
+			args: args{
+				paramType: NONE,
+				path:      "",
+			},
+			wantErr: false,
+			want:    "util",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Log("Running test: ", tt.testName)
+		t.Run(tt.testName, func(t *testing.T) {
+			name, err := GetComponentDir(tt.args.path, tt.args.paramType)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Expected err: %v, but err is %v", tt.wantErr, err)
+			}
+
+			if name != tt.want {
+				t.Errorf("Received name %s which does not match %s", name, tt.want)
+			}
+		})
+	}
+}
+
+func TestTruncateString(t *testing.T) {
+	tests := []struct {
+		testName  string
+		str       string
+		strLength int
+		want      string
+	}{
+		{
+			testName:  "Case: Truncate string to greater length",
+			str:       "qw",
+			strLength: 4,
+			want:      "qw",
+		},
+		{
+			testName:  "Case: Truncate string to lesser length",
+			str:       "rtyu",
+			strLength: 3,
+			want:      "rty",
+		},
+		{
+			testName:  "Case: Truncate string to -1 length",
+			str:       "Odo",
+			strLength: -1,
+			want:      "Odo",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Log("Running test: ", tt.testName)
+		t.Run(tt.testName, func(t *testing.T) {
+			receivedStr := TruncateString(tt.str, tt.strLength)
+			if tt.want != receivedStr {
+				t.Errorf("Truncated string %s is not same as %s", receivedStr, tt.want)
+			}
+		})
+	}
+}
+
+func TestGenerateRandomString(t *testing.T) {
+	tests := []struct {
+		testName  string
+		strLength int
+	}{
+		{
+			testName:  "Case: Generate random string of length 4",
+			strLength: 4,
+		},
+		{
+			testName:  "Case: Generate random string of length 3",
+			strLength: 3,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Log("Running test: ", tt.testName)
+		t.Run(tt.testName, func(t *testing.T) {
+			name := GenerateRandomString(tt.strLength)
+			r, _ := regexp.Compile(fmt.Sprintf("[a-z]{%d}", tt.strLength))
+			match := r.MatchString(name)
+			if !match {
+				t.Errorf("Randomly generated string %s which does not match regexp %s", name, fmt.Sprintf("[a-z]{%d}", tt.strLength))
+			}
+		})
+	}
+}
+
+func TestStringInSlice(t *testing.T) {
```

Comment:
We don't need this test as the function is not used anywhere

Reply:
Ack Done